File: src/server/db/models/File.ts
import { Column, DataType, HasMany,Scopes, Table, } from 'sequelize-typescript';

import { StorageType, } from '../../storages';
import { AbstractModel, } from './abstract/AbstractModel';
import { FileUser, } from './FileUsers';

@Scopes(() => ({
  defaultScope: {
    attributes: {
      exclude: ['createdAt', 'updatedAt'],
    },
  },
}))
@Table({})
export class File extends AbstractModel {
  @Column({
    type: DataType.STRING(10),
    allowNull: false,
  })
    ext!: string;

  @Column({
    type: DataType.STRING,
    allowNull: false,
  })
    mime!: string;

  @Column({
    type: DataType.BIGINT,
    allowNull: false,
  })
    size!: number;

  @Column({
    type: DataType.STRING,
    defaultValue: StorageType.DB,
  })
    storage!: StorageType;

  @Column({
    type: DataType.STRING(),
    allowNull: false,
  })
    hash!: string;

  @HasMany(() => FileUser)
    fileUsers?: FileUser[];
}

File: app/db/models/File.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type File struct {
	ID        uuid.UUID `gorm:"primaryKey;type:uuid;default:uuid_generate_v4()"`
	CreatedAt time.Time
	UpdatedAt time.Time
	EXT       string    `gorm:"type:varchar(10);column:ext;not null"`
	MIME      string    `gorm:"type:varchar(255);column:mime;not null"`
	Size      string    `gorm:"type:bigint;not null"`
	Storage   string    `gorm:"type:varchar(255);not null"`
	Hash      string    `gorm:"type:varchar(255);not null"`
	// fileUsers *FileUser
}
